### PR TITLE
Add Swift Package Manager (SPM) support

### PR DESCRIPTION
**Pull request by jaeopt**
* major changes to the project structures following SPM standard (Sources/Tests)
* tested with Xcode 11 (beta) and confirmed it works with xcode SPM ok
* tested this new structure works ok with CocoaPods as well

### CODE REVIEW

File: OptimizelySwiftSDK.podspec [hunk 1]
```diff
@@ -12,9 +12,7 @@ Pod::Spec.new do |s|
     :git => "https://github.com/optimizely/swift-sdk.git",
     :tag => "v"+s.version.to_s
   }
-  s.source_files            = "OptimizelySDK/**/*.{swift,h}"
-  s.public_header_files     = "OptimizelySDK/OptimizelySwiftSDK/*.h"
```
Comment (thomaszurkan-optimizely) [inline OptimizelySwiftSDK.podspec]: I think we still need to header files and exclude

Comment (thomaszurkan-optimizely) [reply OptimizelySwiftSDK.podspec]: Can we also fix summary and homepage?
"Optimizely server-side testing core framework."
"Optimizely iOS/tvOS experiment framework" or something like that. :)

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        .tvOS(.v9)
+    ],
+    products: [
+        .library(name: "Optimizely",
+                 targets: ["Optimizely"])
+    ],
+    targets: [
+        .target(name: "Optimizely", path: "Sources")
+    ],
+    swiftLanguageVersions: [.v5]
```
Comment (thomaszurkan-optimizely) [inline Package.swift]: Do we want to change to use Swift 5?

File: OptimizelySwiftSDK.podspec (hunk 1 above)
Comment (jaeopt) [reply OptimizelySwiftSDK.podspec]: It looks like header files extracted from the src path by default. We can add the header line when we need support for other paths.

File: Package.swift (hunk 2 above)
Comment (jaeopt) [reply Package.swift]: Yep. Package.swift is already changed to Swift 5. I'll fix others to be consistent (Swift5+).

Comment (jaeopt) [reply Package.swift]: I have tested with CocoaPod, Carthage, and SPM (Xcode 11). Found a couple of issues and fixed. All looks good. I'll make sure again when releasing.

**Pull request merged**